db/db_impl/db_impl_open.cc

```diff
@@ -499,7 +515,9 @@ Status DBImpl::Recover(
     // attention to it in case we are recovering a database
     // produced by an older version of rocksdb.
     std::vector<std::string> filenames;
-    s = env_->GetChildren(immutable_db_options_.wal_dir, &filenames);
+    if (!immutable_db_options_.best_efforts_recovery) {
```

Comment:
Is just ignoring WAL files when best_efforts_recovery is enabled correct? Or should the recovery fail if there are non-empty WAL files?

Reply:
On the one hand, WAL contains data ahead of SSTs (not considering WAL with prepared entries). On the other hand, it is very common to have non-empty WAL even if db closed cleanly.
I think we should not report best-efforts recovery as failure.

Reply:
Furthermore, I think we need to bump up cfd->log_number and min_log_number_to_keep after we ignore the existing WALs.

Reply:
I was thinking about cross-CF consistency. Do we implicitly guarantee it if the user uses WAL? We don't guarantee it with the best efforts recovery right? Maybe we can just mention it in READ.md.

Reply:
Good point. User may very likely assume that cross-CF consistency is guaranteed if they enable WAL. We should call this out in HISTORY.md.